File: src/main/java/com/Johnny/student/Student.java
package com.Johnny.student;

public class Student {
    String id;
    String name ;
    int english;
    int math ;


    public Student(String name, int english, int math) {
        this.name = name;
        this.english = english;
        this.math = math;
    }

    public int highest() {

        /*int max = 0;
        if ( english > math ) {
            max = english ;
        }
        else max = math ;
        */
        return (english > math ) ? english : math ;


    }

    public void print() {
        int average = getAverage() ;
        System.out.print(name + "\t" + english + "\t" + math + "\t" + getAverage() + "\t" +
                (( average >= 60 ) ? "PASS" : "FAILED")); /*三元運算*/
        char gradding = 'F' ;
        /*
        if ( average >= 90 && average <= 100 ) gradding = 'A' ;
        else if ( average >= 80 && average <= 89 ) gradding = 'B' ;
        else if ( average >= 70 && average <= 79 ) gradding = 'C' ;
        else if ( average >= 60 && average <= 69 ) gradding = 'D' ;
        */

        switch(average/10) {
            case 10:
            case 9:
                gradding = 'A' ;
                break ;
            case 8:
                gradding = 'B' ;
                break ;
            case 7:
                gradding = 'C' ;
                break ;
            case 6:
                gradding = 'D' ;
                break ;
            default:
                gradding = 'F' ;
        }

        System.out.println("\t"+ gradding);

       /*
        if ( getAverage() >= 60 ) {
            System.out.println("\tPASS");
        } else System.out.println("\tFAILED");
        */
    }


    public int getAverage() {
        return (english+math)/2 ;
    }
}

File: src/main/kotlin/com/Johnny/student/student.kt
package com.Johnny.kotlin

import java.util.*

fun main() {
    val stu = Student("Johnny", 77, 99)
    stu.print()
    var test = 123
    println( "Test : $test")
    println("Hight score: ${stu.hightest()}" )
}


class Student( var name: String?, var english: Int, var math: Int) {
    fun print() {
        /*
        print( name + "\t" + english + "\t" + math + "\t" + getAverage() + "\t" + passOrFailed() )
        println("\t" + gradding() )
        */
        println("$name\t$english\t$math\t${getAverage()}\t${passOrFailed()}\t${gradding()}")

    }
    fun gradding() = when(getAverage()) {
        //90 -> 'A'
        in 90..100 -> 'A' // scope
        in 80..89 -> 'B'
        in 70..79 -> 'C'
        in 60..69 -> 'D'
        else -> 'F'
    }
    fun passOrFailed() = if ( getAverage() >= 60 ) "PASS" else "FAILED"

    fun getAverage() = (english+math)/2

    fun hightest() = if ( english > math ) {
        println("english")
        english
    }
    else {
        println("math")
        math
    }

    fun nameCheck() = println(name?.length)
}

private fun userInput() {
    val scanner = Scanner(System.`in`)
    print("Please enter student's name: ")
    var name = null // scanner.next()

    var english = scanner.nextInt()
    print("Please enter student's english: ")

    var math = scanner.nextInt()
    print("Please enter student's math: ")

}


